src/app/ReadClient.cpp

```diff
@@ -791,6 +829,16 @@ CHIP_ERROR ReadClient::SendSubscribeRequest(ReadPrepareParams & aReadPreparePara
     request.IsFabricFiltered(aReadPrepareParams.mIsFabricFiltered).EndOfSubscribeRequestMessage();
     ReturnErrorOnFailure(err = request.GetError());
 
+    if (aReadPrepareParams.mAttributePathParamsListSize != 0 && aReadPrepareParams.mpAttributePathParamsList != nullptr &&
+        aReadPrepareParams.mDataVersionFilterParamsListSize != 0 && aReadPrepareParams.mpDataVersionFilterParamsList != nullptr)
```

Comment:
Can we not duplicate this logic between read and subscribe?  Why isn't it all in the shared function?

Reply:
see request, one request is ReadRequest, another is subscribeRequest, different but similar class in lots of angle, it seems it is not easy subscribe request inherit from read request in Message's encoding/decoding perspective, here  if I put them in same shared function, it seems it would not work?

Reply:
Ah, ok.  If the request things are different types sharing is going to be hard, yes.